### PR TITLE
Support private subnets and existing S3 buckets for self-hosted storage

### PR DESCRIPTION
**Pull request by yolken-airplane**
## Description
This change updates our AWS terraform configs to support providing an extra set of subnet ids for the self-hosted storage ALB so that the agent itself can stay private. It also allows users to "bring their own bucket" if they want.

cc: @zhan-airplane 

## Testing
Testing completed successfully via applying locally.

### CODE REVIEW

File: modules/storage/README.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ See the [Airplane docs](https://docs.airplane.dev/self-hosting/storage) for more
 | <a name="input_team_id"></a> [team\_id](#input\_team\_id) | Airplane team ID - retrieve via `airplane auth info` | `string` | n/a | yes |
 | <a name="input_agent_iam_role_name"></a> [agent\_iam\_role\_name](#input\_agent\_iam\_role\_name) | Name of an existing IAM role to use for the agent. If blank, then a new IAM role will be created. | `string` | `""` | no |
 | <a name="input_agent_storage_domain"></a> [agent\_storage\_domain](#input\_agent\_storage\_domain) | For development purposes only | `string` | `"d.airplane.sh"` | no |
+| <a name="input_agent_storage_s3_bucket"></a> [agent\_storage\_s3\_bucket](#input\_agent\_storage\_s3\_bucket) | Name of an existing bucket to use with for self-hosted storage. If blank, a new bucket will be created. | `string` | `""` | no |
```
Comment (ameyakhare) [inline modules/storage/README.md:51]:
```suggestion
| <a name="input_agent_storage_s3_bucket"></a> [agent\_storage\_s3\_bucket](#input\_agent\_storage\_s3\_bucket) | Name of an existing bucket to use for self-hosted storage. If blank, a new bucket will be created. | `string` | `""` | no |
```

File: modules/storage/variables.tf [hunk 2]
```diff
@@ -15,6 +15,12 @@ variable "agent_storage_domain" {
   default     = "d.airplane.sh"
 }
 
+variable "agent_storage_s3_bucket" {
+  type        = string
+  description = "Name of an existing bucket to use with for self-hosted storage. If blank, a new bucket will be created."
```
Comment (ameyakhare) [inline modules/storage/variables.tf:20]:
```suggestion
  description = "Name of an existing bucket to use for self-hosted storage. If blank, a new bucket will be created."
```

File: variables.tf [hunk 3]
```diff
@@ -41,12 +41,24 @@ variable "agent_storage_domain" {
...
 }
 
+variable "agent_storage_s3_bucket" {
+  type        = string
+  description = "Name of an existing bucket to use with for self-hosted storage. If blank, a new bucket will be created."
```
Comment (ameyakhare) [inline variables.tf:58]:
```suggestion
  description = "Name of an existing bucket to use for self-hosted storage. If blank, a new bucket will be created."
```

**Review by ameyakhare — APPROVED** [review]

File: variables.tf (hunk 3 above)
Comment (yolken-airplane) [reply variables.tf]: Fixed.

File: modules/storage/variables.tf (hunk 2 above)
Comment (yolken-airplane) [reply modules/storage/variables.tf]: Fixed.

File: modules/storage/README.md (hunk 1 above)
Comment (yolken-airplane) [reply modules/storage/README.md]: Fixed.

**Pull request merged**